redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt

```diff
@@ -0,0 +1,216 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.layout.uiview
+
+import app.cash.redwood.Modifier
+import app.cash.redwood.layout.api.Constraint
+import app.cash.redwood.layout.api.CrossAxisAlignment
+import app.cash.redwood.layout.modifier.HorizontalAlignment
+import app.cash.redwood.layout.modifier.VerticalAlignment
+import app.cash.redwood.layout.widget.Box
+import app.cash.redwood.ui.Dp
+import app.cash.redwood.ui.Margin
+import app.cash.redwood.ui.dp
+import app.cash.redwood.widget.UIViewChildren
+import kotlinx.cinterop.CValue
+import kotlinx.cinterop.convert
+import kotlinx.cinterop.readValue
+import kotlinx.cinterop.useContents
+import platform.CoreGraphics.CGFloat
+import platform.CoreGraphics.CGRectMake
+import platform.CoreGraphics.CGRectZero
+import platform.CoreGraphics.CGSize
+import platform.CoreGraphics.CGSizeMake
+import platform.UIKit.NSLayoutConstraint
+import platform.UIKit.UIColor
+import platform.UIKit.UIView
+import platform.darwin.NSInteger
+import platform.posix.INT_MAX
+
+internal class UIViewBox : Box<UIView> {
+  override val value: View = View()
+
+  override var modifier: Modifier = Modifier
+
+  override val children = value.children
+
+  override fun width(width: Constraint) {
+    value.widthConstraint = width
+  }
+
+  override fun minWidth(minWidth: Dp) {
+    value.minWidth = minWidth
+  }
+
+  override fun maxWidth(maxWidth: Dp) {
+    value.maxWidth = maxWidth
+  }
+
+  override fun height(height: Constraint) {
+    value.heightConstraint = height
+  }
+
+  override fun minHeight(minHeight: Dp) {
+    value.minHeight = minHeight
+  }
+
+  override fun maxHeight(maxHeight: Dp) {
+    value.maxHeight = maxHeight
+  }
+
+  override fun margin(margin: Margin) {
+  }
+
+  override fun horizontalAlignment(horizontalAlignment: CrossAxisAlignment) {
+    value.horizontalAlignment = horizontalAlignment
+  }
+
+  override fun verticalAlignment(verticalAlignment: CrossAxisAlignment) {
+    value.verticalAlignment = verticalAlignment
+  }
+
+  internal class View() : UIView(CGRectZero.readValue()) {
+    var widthConstraint = Constraint.Wrap
+    var heightConstraint = Constraint.Wrap
+
+    var minWidth = 0.dp
+    var maxWidth = INT_MAX.dp
+
+    var minHeight = 0.dp
+    var maxHeight = INT_MAX.dp
+
+    var horizontalAlignment = CrossAxisAlignment.Start
+    var verticalAlignment = CrossAxisAlignment.Start
+
+    val children = UIViewChildren(
+      this,
+      insert = { view, index ->
+        insertSubview(view, index.convert<NSInteger>())
+        view.setNeedsLayout()
+      },
+      remove = { index, count ->
+        val views = Array(count) {
+          typedSubviews[index].also(UIView::removeFromSuperview)
```

Comment:
nit: just to be safe, it can't hurt to bail early if `index >= typedSubviews.size`, and maybe throw an error rather than crashing

Reply:
Generally our strategy has been to explode loudly on unexpected input. It gives bugs fewer places to hide!